### PR TITLE
[QN] Fix validator verification

### PR DESCRIPTION
**Pull request by thesan**
- #4824

based on #5057

### CODE REVIEW

File: tests/network-tests/src/fixtures/membership/VerifyValidatorMembershipFixture.ts [hunk 1]
```diff
@@ -49,22 +50,38 @@ export class VerifyValidatorMembershipFixture extends WithMembershipWorkersFixtu
...
 
   async runQueryNodeChecks(): Promise<void> {
     await super.runQueryNodeChecks()
-    const qMembers = await this.query.getMembersByIds(this.verifications.map((m) => m.memberId))
-    this.assertQueriedMembershipsAreValid(qMembers)
+
+    const memberIds = this.verifications.reduce(
+      (ids, { memberId }) => (ids.some((id) => memberId.eq(id)) ? ids : [...ids, memberId]),
+      [] as MemberId[]
+    )
```
Comment (zeeshanakram3) [inline tests/network-tests/src/fixtures/membership/VerifyValidatorMembershipFixture.ts]: Don't you think following would be much simpler 
```suggestion
    const memberIds = [...new Set(this.verifications.map((v) => v.memberId))]
```

**Review by zeeshanakram3 — CHANGES_REQUESTED** [review]
Look good, while testing this PR I found a logical bug in the `verifyValidator` CLI command which I overlooked in the PR review.

https://github.com/Joystream/joystream/blob/master/cli/src/commands/working-groups/verifyValidator.ts#L32

It should be the following 
```ts
 if (!worker || this.group !== WorkingGroups.Membership) {
```

File: tests/network-tests/src/fixtures/membership/VerifyValidatorMembershipFixture.ts (hunk 1 above)
Comment (thesan) [reply tests/network-tests/src/fixtures/membership/VerifyValidatorMembershipFixture.ts]: :100:

**Comment by thesan** [comment]
> ```ts
>  if (!worker || this.group !== WorkingGroups.Membership) {
> ```

:sweat: 
Nice catch!

**Review by zeeshanakram3 — APPROVED** [review]
LGTM

**Pull request merged**